### PR TITLE
Add Markdownlint

### LINKED ISSUE #1: Create formatting guidelines for markdown and enforce with GitHub Actions
Consider how [OCI Image spec](https://github.com/opencontainers/image-spec/blob/main/Makefile#L96-L98) this.

### PR DESCRIPTION
**Pull request by ktarplee**
Closes #1 

Adds the GitHub action to run Markdown lint on all PRs.  This MR also fixes the issues found from the CNCF template that did not pass markdownlint.

We still need to determine if we need a style guide for the Markdown files.

### CODE REVIEW

**Comment by ktarplee** [comment]
I just noticed #4 was already open.

**Comment by afflom** [comment]
Thank you for submitting this @ktarplee. I defer to a gh actions expert to compare this to #4, but I see that your solution already fixes the found formatting discrepancies from the repo, which is convenient.

File: .github/workflows/testing.yml
Comment (rchincha) [inline .github/workflows/testing.yml]: Thanks for this. The only comment would whichever is better maintained.

Comment (ktarplee) [reply .github/workflows/testing.yml]: I switched to the same one you are using `DavidAnson/markdownlint-cli2-action`.  It does seem to be better since it runs faster because it does not use docker to build an image.

**Comment by sudo-bmitch** [comment]
Have GHA been enabled on this repo? By default on forks they are disabled, and this repo is a fork of the template repo. A maintainer needs to flip the switch for this to work.

File: .github/workflows/testing.yml
Comment (sudo-bmitch) [inline .github/workflows/testing.yml]:
```suggestion
on:
  push:
  pull_request:
    branches: [ main ]
```

**Review by sudo-bmitch — COMMENTED** [review]
We need to adjust the GHA to run on pull requests.

File: .github/workflows/testing.yml
Comment (ktarplee) [reply .github/workflows/testing.yml]: I tried that but still no luck. @sudo-bmitch

Comment (sudo-bmitch) [reply .github/workflows/testing.yml]: At this point, I'm out of ideas other than the maintainers checking things on their side or merging and see if that helps.

Comment (ktarplee) [reply .github/workflows/testing.yml]: Maybe they have to merge this and then subsequent PRs will be impacted properly???  @afflom

**Review by afflom — APPROVED** [review]
@rchincha I concur with @ktarplee. Are we good to merge this?

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**
